File: 135-candy/candy.py
class Solution:
    def candy(self, ratings: List[int]) -> int:
        n = len(ratings)
        candies = [1] * n

        # Left to right pass
        for i in range(1, n):
            candies[i] = candies[i - 1] + 1 if ratings[i] > ratings[i - 1] else 1

        # Right to left pass
        for i in range(n - 2, -1, -1):
            candies[i] = max(candies[i], candies[i + 1] + 1) if ratings[i] > ratings[i + 1] else candies[i]

        return sum(candies)    

File: 135-candy/candy.js
/**
 * @param {number[]} ratings
 * @return {number}
 */
var candy = function(ratings) {
    const n = ratings.length;
    const candies = new Array(n).fill(1);

    for (let i = 1; i < n; i++) {
        candies[i] = ratings[i] > ratings[i - 1] ? candies[i - 1] + 1 : 1;
    }

    for (let i = n - 2; i >= 0; i--) {
        candies[i] = Math.max(candies[i], ratings[i] > ratings[i + 1] ? candies[i + 1] + 1 : 1);
    }

    return candies.reduce((a, b) => a + b, 0);

};